tests/gtest/avifrgbtoyuvtest.cc

```diff
@@ -839,5 +839,62 @@ INSTANTIATE_TEST_SUITE_P(
                    AVIF_CHROMA_UPSAMPLING_BILINEAR),
             /*has_alpha=*/Bool()));
 
+class SetGetRGBATest
+    : public testing::TestWithParam<std::tuple<
+          /*rgb_depth=*/int, avifRGBFormat, /*is_float=*/bool>> {};
+
+TEST_P(SetGetRGBATest, SetGetTest) {
+  const int rgb_depth = std::get<0>(GetParam());
+  const avifRGBFormat rgb_format = std::get<1>(GetParam());
+  const bool is_float = std::get<2>(GetParam());
+
+  // Unused yuv image, simply needed to initialize the rgb image.
+  testutil::AvifImagePtr yuv(avifImageCreate(4, 2, 8, AVIF_PIXEL_FORMAT_YUV444),
+                             avifImageDestroy);
+
+  testutil::AvifRgbImage rgb(yuv.get(), rgb_depth, rgb_format);
+  rgb.isFloat = is_float;
+
+  avifRGBColorSpaceInfo color_space;
+  if (!avifGetRGBColorSpaceInfo(&rgb, &color_space)) {
+    return;  // Invalid combination e.g. RGB565 with depth > 8.
```

Comment:
I recommend tweaking the `INSTANTIATE_TEST_SUITE_P` below rather than keeping this check that will silently skip invalid tests. For example I doubt that `/*is_float=*/Values(true)` works with `/*rgb_depth=*/Values(8)`.

There could also be negative checks to verify that `avifGetRGBColorSpaceInfo()` returns false when expected, but this is unnecessary in my opinion.

Reply:
Urgh you're right I meant to use 16 for the floats :/ Thanks for catching this.